src/app/search/SEARCH.worker.ts

```
@@ -29,30 +30,39 @@ export interface WorkerMessageOutputSearchResult extends WorkerMessageBase {
-// TODO: Come up with some method of cache invalidation. Ideally, the layout component would calculate the hash of the search index and pass it to the client
-const searchIndexPromise: Promise<SearchIndexEntity[]> = fetch(new URL(`/${toLowerCase(game)}/search-index.json`, self.location.href), {
+const searchIndexPromise: Promise<SearchIndexEntity[]> = fetch(new URL(`/${toLowerCase(game)}/search-index.json?hash=${searchIndexHash}`, self.location.href), {
     cache: 'force-cache',
-}).then(res => res.json());
+}).then(async res => {
+    if (!res.ok) {
+        try {
+            console.error('[SEARCH WORKER] Failed to fetch search index:', res.status, res.statusText, 'with result:', await res.text());
+        } finally {
+            // eslint-disable-next-line no-unsafe-finally
+            throw new Error(`[SEARCH WORKER] Failed to fetch search index: ${res.status} ${res.statusText}`);
```

Biome (1.9.4)

[error] Unsafe usage of 'throw'.

'throw' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)